configure.ac

```
@@ -124,9 +139,51 @@ fi
 AM_CONDITIONAL(ENABLE_PANELAPPLET, test "x$enable_panelapplet" != "xno")
 
 #============================================================================
-# Sanity check: must build either status icon or panel applet
+# applet in-process and Wayland support
 #============================================================================
 
+AC_ARG_ENABLE([wayland],
+              [AS_HELP_STRING([--enable-wayland],
+                              [Explicitly enable or disable Wayland support
+                               (default is to enable only if Wayland client development library is detected)])],
+              [enable_wayland=$enableval],
+              [enable_wayland=auto])
+AC_ARG_ENABLE([in-process],
+              [AS_HELP_STRING([--enable-in-process],
+                              [Enable in-process build of panel applet (default: no, required for wayland)])],
+              [enable_in_process=$enableval],
+              [dnl the default follows whether Wayland is requested or not,
+               dnl as Wayland support requires it but we d rather not have it
+               dnl otherwise for the moment
+               enable_in_process=$enable_wayland])
+
+dnl $enable_in_process is gonna be either:
+dnl - "no" if explicitly disabled
+dnl - "yes" if explicitly enabled or if Wayland is explicitly enabled
+dnl - "auto" if nothing specific is requested, in which case, default back to "no"
+AS_IF([test "x$enable_in_process" != xyes], [enable_in_process=no])
+
+WAYLAND_DEPS="gtk-layer-shell-0 wayland-client gdk-wayland-3.0"
```

So if needed, move the version checks here
```suggestion
WAYLAND_DEPS="gtk-layer-shell-0 >= $GTK_LAYER_SHELL_REQUIRED_VERSION wayland-client gdk-wayland-3.0 >= $GDK_WAYLAND_REQUIRED_VERSION"
```

---

This works, and this is the results on Debian Unstable:
gtk-layer-shell-dev not installed, `--enable-wayland` not passed: successfull configuration and build out of process

gtk-layer-shell-dev installed, `--enable-wayland` passed: successfull configuration and build in-process w wayland support

 gtk-layer-shell-dev installed `--disable-wayland --enable-in process`  showed yes for in-process and no for wayland
and this built without issue

and

gtk-layer-shell-dev installed `--enable-wayland --disable-in process` 
errored out as intended with 
`configure: error: --enable-wayland requires --enable-in-process`

---

Thanks for catching this before it got pushed to master